src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestNode.cs

```diff
@@ -85,11 +88,57 @@ public HttpBindingResult<HttpRequestMessage> TryGetHttpRequestMessage(HttpBindin
         }
         else
         {
-            success = false;
             diagnostics.AddRange(uriBindingResult.Diagnostics);
         }
 
-        if (success)
+        var headers =
+            HeadersNode?.HeaderNodes.Select(h => new KeyValuePair<string, string>(h.NameNode.Text, h.ValueNode.Text)).ToArray()
+            ??
+            Array.Empty<KeyValuePair<string, string>>();
+
+        foreach (var kvp in headers)
+        {
+            switch (kvp.Key.ToLowerInvariant())
+            {
+                case "content-type":
+                    if (request.Content is null)
+                    {
+                        request.Content = new StringContent("");
+                    }
+
+                    request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse(kvp.Value);
+                    break;
+                case "accept":
+                    request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse(kvp.Value));
+                    break;
+                case "user-agent":
+                    request.Headers.UserAgent.Add(ProductInfoHeaderValue.Parse(kvp.Value));
+                    break;
+                default:
+                    request.Headers.Add(kvp.Key, kvp.Value);
```

Comment:
minor: It is surprising that this call would not internally handle `content-type`, `accept` and `user-agent` cases above. Are these the only three header kinds that callers are expected to handle separately?

Reply:
The headers grammar is conceptually flat but the `HttpRequestMessage` object model spreads them across these different child objects. There might be other cases that require special treatment. This is probably also something we'll need to take a systematic approach to in the next rounds of testing, like the grammar combinations.